### PR TITLE
feat(vscode): use CommandPalette to display tabby menu

### PR DESCRIPTION
**Pull request by wsxiaoys**
image

cc @liangfung

### CODE REVIEW

File: clients/vscode/src/CommandPalette.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import { commands, QuickPick, QuickPickItem, QuickPickItemKind, ThemeIcon, window } from "vscode";
+import { State as LanguageClientState } from "vscode-languageclient";
+import { Client } from "./lsp/Client";
+import { Config } from "./Config";
+
+export default class CommandPalette {
+  quickPick: QuickPick<CommandPaletteItem>;
+  isChatEnabled = false;
```
Comment (wsxiaoys) [inline clients/vscode/src/CommandPalette.ts:8]:
```suggestion
```

File: clients/vscode/src/CommandPalette.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    private readonly client: Client,
+    private readonly config: Config,
+  ) {
+    this.isChatEnabled = this.client.chat.isAvailable;
+    this.quickPick = window.createQuickPick();
+    this.quickPick.title = "Tabby Command Palette";
+
+    let items: CommandPaletteItem[] = [this.itemForStatus()];
+
+    if (this.isChatEnabled) {
```
Comment (wsxiaoys) [inline clients/vscode/src/CommandPalette.ts:20]:
```suggestion
    if (this.client.chat.isAvailable) {
```

File: clients/vscode/src/Commands.ts [hunk 3]
```diff
@@ -165,35 +167,27 @@ export class Commands {
...
           },
         ])
         .then((selection) => {
           if (selection) {
             switch (selection.label) {
-              case "Online Documentation":
-                env.openExternal(Uri.parse("https://tabby.tabbyml.com/"));
-                break;
               case "Model Registry":
                 env.openExternal(Uri.parse("https://tabby.tabbyml.com/docs/models/"));
```
Comment (icycodes) [inline clients/vscode/src/Commands.ts:192]:
```suggestion
              case "Website":
                env.openExternal(Uri.parse("https://tabby.tabbyml.com/"));
```

File: clients/vscode/src/StatusBarItem.ts [hunk 4]
```diff
@@ -181,11 +160,6 @@ export class StatusBarItem {
     this.item.backgroundColor = backgroundColorWarning;
     this.item.text = `${iconDisconnected} ${label}`;
     this.item.tooltip = "Cannot connect to Tabby Server. Click to open settings.";
-    this.item.command = {
-      title: "",
-      command: "tabby.applyCallback",
-      arguments: [() => this.issues.showHelpMessage("connectionFailed")],
-    };
```
Comment (icycodes) [inline clients/vscode/src/StatusBarItem.ts:188]: There will be no button to show help message after this change. Is it expected?

Comment (wsxiaoys) [reply clients/vscode/src/StatusBarItem.ts:188]: it'll open settings page and show `Disconnected` in palette

image

**Pull request merged**